File: Projects/08/TempTable.rb
#############################################
# Emily Costa
# 4/14/2020
# Project 8
# TempTable.rb 
#############################################
load 'Temperature.rb'

# make new Temperature object
def make()
	print "Enter temperature: "
	inputString = gets
	temp = inputString.split(' ')
	temp1 = Temperature.new(temp[0],temp[1])
end

def main
	# scan in values
	print "\nbaseTemp -> "
	baseTemp = make()
	print "limitTemp -> "
	limitTemp = make()
	print "stepValue -> Enter value: "
	stepValue = gets.to_f

	temp1 = baseTemp;
	temp2 = baseTemp;

	# loop to print table
	if (baseTemp.equals(baseTemp,limitTemp) == true)
		convertTempF = baseTemp.convertF()
		convertTempC = baseTemp.convertC()
		convertTempK = baseTemp.convertK()
		print "#{convertTempF.print}\t#{convertTempC.print}\t#{convertTempK.print}\n"
		
	elsif (baseTemp.less_than(baseTemp,limitTemp) == true)
		while (baseTemp.less_than(temp1,limitTemp) == true) do
			convertTempF = temp1.convertF()
			convertTempC = temp1.convertC()
			convertTempK = temp1.convertK()
			print "#{convertTempF.print}\t#{convertTempC.print}\t#{convertTempK.print}\n"
			temp1 = temp2.raise(stepValue)
			temp2 = temp1
		end 
		convertTempF = limitTemp.convertF()
		convertTempC = limitTemp.convertC()
		convertTempK = limitTemp.convertK()
		print "#{convertTempF.print}\t#{convertTempC.print}\t#{convertTempK.print}\n"

	elsif (baseTemp.less_than(baseTemp,limitTemp) == false)
		while (baseTemp.less_than(temp1,limitTemp) == false) do
			convertTempF = temp1.convertF()
			convertTempC = temp1.convertC()
			convertTempK = temp1.convertK()
			print "#{convertTempF.print}\t#{convertTempC.print}\t#{convertTempK.print}\n"
			temp1 = temp2.lower(stepValue)
			temp2 = temp1
		end 
		convertTempF = limitTemp.convertF()
		convertTempC = limitTemp.convertC()
		convertTempK = limitTemp.convertK()
		print "#{convertTempF.print}\t#{convertTempC.print}\t#{convertTempK.print}\n"
	end

end # end main

main

File: Projects/08/TempTable.clj
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; TempTable.clj uses Temperature name type in Tmeperature.clj
;;; Emily Costa
;;; 4/14/2020
;;; Project 08
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns TempTable)

(use '[clojure.string :only (join split)])
(load-file "src/Temperature.clj")

(defn -main []
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; scan in input
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
	(print "\nbaseTemp -> ") (flush)
	(let [
		degree (read-line)
		deg (split degree #" ")
		degree (Float/parseFloat (get deg 0))
		scale (get deg 1) 
		baseTemp (make-Temp degree scale)
	]

		(print "limitTemp -> ") (flush)
		(let [
			degree (read-line)
			deg (split degree #" ")
			degree (Float/parseFloat (get deg 0))
			scale (get deg 1) 
			limitTemp (make-Temp degree scale)
		]

			(print "stepValue -> ") (flush)
			(let [
				stepValue (read)
			]

			(def temp1 baseTemp)
			(def temp2 baseTemp)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; if statements to print table
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
			; if base and limit are equal
			(if (true? (equals baseTemp limitTemp))
				(do
					(printTemp (convertF limitTemp)) (print "\t\t")
					(printTemp (convertC limitTemp)) (print "\t\t")
					(printTemp (convertK limitTemp)) 
					(print " IN EQUALS")
				) ; end do
				(do

				; if base is less than limit
					(if (true? (less-than baseTemp limitTemp))
					(do
						(while (true? (less-than temp1 limitTemp))
							(do
								(printTemp (convertF temp1)) (print "\t\t")
								(printTemp (convertC temp1)) (print "\t\t")
								(printTemp (convertK temp1)) (print "\n")
								(def temp1 (raise temp2 stepValue))
								(def temp2 temp1)
								)
							) ; end do
						) ; end while
					) ; end if

					; if base if greater than limit
					(if (false? (less-than baseTemp limitTemp))
					(do
						(while (false? (less-than temp1 limitTemp))
							(do
								(printTemp (convertF temp1)) (print "\t\t")
								(printTemp (convertC temp1)) (print "\t\t")
								(printTemp (convertK temp1)) (print "\n")
								(def temp1 (lower temp2 stepValue))
								(def temp2 temp1)
								)
							) ; end do
						) ; end while
					) ; end if


				) ; end do
			) ; end if

			(print "\n")

			) ; end let
		) ; end let
	) ; end let
)